### ISSUE TITLE
Segmentation fault when use call_user_func with dynamic methods

### ISSUE THREAD

**Issue by noldor** [krakjoe/uopz#137]
uopz: master
PHP 8.0.1 (cli) (built: Jan 11 2021 00:57:45) ( NTS DEBUG GCOV )
Copyright (c) The PHP Group
Zend Engine v4.0.1, Copyright (c) Zend Technologies
    with Xdebug v3.0.2, Copyright (c) 2002-2021, by Derick Rethans
    
GDB output
```
GNU gdb (Ubuntu 9.2-0ubuntu1~20.04) 9.2
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
--Type <RET> for more, q to quit, c to continue without paging--RET
Reading symbols from php...
(gdb) run test.php
Starting program: /home/noldor/.phpbrew/php/php-8.0.1/bin/php test.php
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff4297a94 in uopz_find_hook (function=0x0) at /home/noldor/.phpbrew/build/php-8.0.1/ext/uopz/src/hook.c:128
128             if (!function->common.function_name) {
```

How to reproduce:

```php
<?php

declare(strict_types=1);

class Test {
    public static function __callStatic($name, $args)
    {
        return 123;
    }

    public function __call($name, $args)
    {
        return 123;
    }
}

call_user_func_array([new Test, 'test'], [1, 2, 3]);
call_user_func_array([Test::class, 'test'], [1, 2, 3]);
call_user_func([new Test, 'test'], 1, 2, 3);
call_user_func([Test::class, 'test'], 1, 2, 3);
```

Any call to `call_user_func` or `call_user_func_array` will produce segmentation fault.

Can you fix this, please?

**Comment by keywinf** [comment]
@noldor Is it the only segfault you encountered so far? Also having trouble these days

**Comment by noldor** [comment]
@keywinf other segfalts did not ccured. I install these branc https://github.com/Articus/uopz/tree/fix-segfault-on-dyn-func-call with phpbrew and it works well.

**Comment by keywinf** [comment]
Yeah I've seen it, I've done a fork with the changes of Articus inside, but still getting different segfaults besides (different reasons).

Thanks anyway!

**Comment by krakjoe** [comment]
Fixed in c50aba975510df783e792bff1a86f9b11e2a288c
